### PR TITLE
Rebrand (pre-RC)

### PR DESCRIPTION
**Pull request by aanand**
Replaces #806.

Future versions of Fig are going to be rebranded to fit alongside the suite of Docker orchestration tools (the other ones currently being Machine and Swarm). The new name is Compose.

A number of things need to be done to make this happen:

- [x] Rename the `fig` Python module to `compose`
- [x] Rename `fig.yml` to `compose.yml`
- [x] Deprecate `FIG_*` environment variables, introduce `COMPOSE_*` equivalents
- [x] Rename the `fig` executable
  - [x] Choose a name (chosen: `docker-compose`)
  - [x] Rename Python stub at `bin/fig`
  - [x] Rename binaries built by `script/build-osx` and `script/build-linux`
  - [x] Update the Dockerfile
- [x] Update name in docs, including all instances of:
  - [x] `fig` executable
  - [x] `fig.yml`
  - [x] “Fig”
- [x] Remove all website-related stuff from docs
  - [x] Jekyll tooling (Dockerfile, `fig.yml`, build script)
  - [x] CSS
  - [x] Images
  - [x] Markup
- [x] Update name in `README.md`, `ROADMAP.md` and `CONTRIBUTING.md`
- [x] Remove rebranding plans from `ROADMAP.md` 
- [x] Check that we haven’t messed up the word “config” (or any of its derivatives) anywhere
- [x] Create a new PyPi package called `docker-compose`

### CODE REVIEW

**Comment by bfirsh** [comment]
Looks good! We probably can't call the binary "compose". Could we add a to-do item for choosing a name for the binary, so this does not look complete if you rename it to "compose"?

**Comment by dnephin** [comment]
Maybe `docker-compose` for the pypi package?

Another task might be: Update IRC channel in the docs to `#docker-compose` (and register the new channel) ?

**Comment by aanand** [comment]
@bfirsh @dnephin done.

**Comment by bfirsh** [comment]
We also need to make it very clear in the documentation that "Compose" was called "Fig".

**Comment by aanand** [comment]
Are we really happy to redirect people from www.fig.sh? If people continue to use Fig for legacy reasons, I don't want to take their documentation away.

Hence "Add a deprecation warning to all pages on www.fig.sh" - I'm thinking a banner at the top saying "Fig has been deprecated and replaced by Docker Compose (link to docs.docker.com)".

**Comment by bfirsh** [comment]
:+1: sounds good

**Comment by bfirsh** [comment]
This plan should probably also take into account that we're doing an official RC release.

E.g. we should be tagging and releasing 1.1.0-rc1

**Comment by lePoxy** [comment]
Any due date for the 1.1.0 ? I was hopping a new release of fig in order to use envfile and dns-search option, but with the rebrand this will take maybe longer.

**Comment by aanand** [comment]
Rebased, build is green.

File: compose/cli/command.py [hunk 1]
```diff
@@ -102,13 +110,13 @@ def get_config_path(self, file_path=None):
         if file_path:
             return os.path.join(self.base_dir, file_path)
 
-        if os.path.exists(os.path.join(self.base_dir, 'fig.yaml')):
-            log.warning("Fig just read the file 'fig.yaml' on startup, rather "
-                        "than 'fig.yml'")
-            log.warning("Please be aware that fig.yml the expected extension "
+        if os.path.exists(os.path.join(self.base_dir, 'docker-compose.yaml')):
+            log.warning("Fig just read the file 'docker-compose.yaml' on startup, rather "
```
Comment (dnephin) [inline compose/cli/command.py]: "Compose just read the file..." ?

Comment (dnephin) [inline compose/cli/command.py]: A fallback to `fig.yml` would be good too

File: compose/project.py [hunk 2]
```diff
@@ -67,7 +67,7 @@ def from_config(cls, name, config, client):
         dicts = []
         for service_name, service in list(config.items()):
             if not isinstance(service, dict):
-                raise ConfigurationError('Service "%s" doesn\'t have any configuration options. All top level keys in your fig.yml must map to a dictionary of configuration options.' % service_name)
+                raise ConfigurationError('Service "%s" doesn\'t have any configuration options. All top level keys in your compose.yml must map to a dictionary of configuration options.' % service_name)
```
Comment (dnephin) [inline compose/project.py]: This mentions `compose.yml` , but the name is actually `docker-compose.yml`, right ?

File: script/clean [hunk 3]
```diff
@@ -1,3 +1,3 @@
 #!/bin/sh
 find . -type f -name '*.pyc' -delete
-rm -rf docs/_site build dist fig.egg-info
+rm -rf docs/_site build dist compose.egg-info
```
Comment (dnephin) [inline script/clean]: the egg file will be `docker-compose.egg-info` I think

**Comment by dnephin** [comment]
merging in-flight branches is going to be some work after this, but not much to be done about that

**Comment by aanand** [comment]
Typos fixed.

**Comment by dnephin** [comment]
LGTM

**Comment by aanand** [comment]
rebranding_0

**Comment by albers** [comment]
I will take care of rebranding the bash completion very soon.

**Comment by Starefossen** [comment]
What about:
* repo name (dicker/fig => docker/compose)
* domain name (fig.sh => compose.sh) 
* maybe add a note in README.md about "previously known as fig"

**Comment by aanand** [comment]
- We'll rename the repo for the stable release.
- There won't be a separate website for Compose - its docs will be rolled into docs.docker.com. www.fig.sh will stay up, and we'll add a banner to the top pointing people to the Compose docs.
- Good idea.

**Comment by albers** [comment]
Bash completion is rebranded, see #871.

**Comment by aanand** [comment]
@lePoxy We've released [1.1.0-rc1](https://github.com/docker/fig/releases/tag/1.1.0-rc1) - give it a whirl.

**Comment by lePoxy** [comment]
@aanand Saw it the day you released it, thanks :)

**Comment by Starefossen** [comment]
Hmm... `fig` was a lot shorter to type than `docker-compose`. Why couldn't the binary at least be `compose`?

**Pull request merged**